RickLeinecker/Fortuna
# Email Confirmation Implemented

**Description**
Email Confirmation on Account Creation implemented.

Routes are in the form:
To Confirm Email:
**localhost:3001/api/user/confirmEmail**: where the body contains:
- the _email address_ to verify
- the _token_ from the email that was sent.

To Resend Confirm Email:
**localhost:3001/api/user/resendConfirm**: where the body contains:
- the _email address_ to check if it still needs to be verified.

Important things that need to be noted with this PR:
- **User Login takes in Email and Password now**: Front-end needs to be updated as such.
- **API functions such that user should not be able to login unless email is verified**: Obviously front-end hasn't been locked down like that, but it needs to be mentioned here.
- **Currently the link sent in the email doesn't go anywhere**: I would like to link it to the thing below.
- **Front-end page for email verification yet to be implemented**: One of the reasons why it took this long to get PR up was that I was trying (and failing) to do it myself. Lols....

If it's desired, I wouldn't mind the updates to front-end be added into this PR, otherwise we should make issues for those once this is merged. I can elaborate on what things should be expected for those new pages.

**I also started utilizing a .env file in this PR. If this is merged, everyone will need the file inside the Fortuna/ directory. It's currently online in the Google Drive.** The amount of use it's been given isn't enough to count that one issue with environment variables being done.

Resolves #16 

**Type of change**
Check options that are relevant.


**Steps to Verify Functionality**
_This will only be the "Golden Path" to verify things work._
1. Checkout branch
2. Download .env file from Drive and place in Fortuna/ directory
3. `npm install`
4. `npm run devserver`
5. Open Postman and MongoDB Compass
6. Register User in Postman (use an actual email or make one in https://ethereal.email/)
    6a. Confirm email sent prompt message is there
    6b. Confirm email was received in your email box
7. Confirm User created in MongoDB Compass with isVerified property as `false`
8. Try Login User via API
    8a. Confirm prompt that user is not verified
9. Try Resending Confirm API (details at the top)
    9a. Confirm prompt that email was sent
    9b. Confirm resend email was received.
10. Using the token at the end of the link in the email (stuff after "ConfirmEmail/"), Confirm the email using the Confirm Email API (details at top)
    10a. Confirm the prompt saying it was verified.
11. Confirm User in MongoDB Compass is verified by checking if the isVerified property is `true`.
12. (Optional) Try running either the Confirm Email API or Resend Email API with respective bodies
    12a. Confirm prompt saying user is verified
13. Login User via API
    13a. Confirm JWT is returned.

There are other ways to test it, and while I've done it myself already, you can feel free to see if there is anything I've missed.

**Final Checklist:**
Go down the list and check them off.
